doc/AI-requirements.adoc

```diff
@@ -0,0 +1,170 @@
+dimensional vector. This will probably not be used in this project.
+
+=== Actions
+
+This is calculated by the neuronal network and will be one decision for a character during fighting in this context.
+The type is going to be a integer in `[0:n]`, where `n` is the number of choices.
+
+== Unity ML-Agents Components
+
+To integrate ML-Agents into a unity project it needs two components.
+
+=== Academy
+
+This is mostly relevant for training. It will setup an environment for the training process. It will be controlled from
+`BrainAcademy` and should have no purpose during production.
+
+=== Agent
+
+The `Agent` _(maybe differently called in ML-Agents 1.0)_ is component of the NPC. It collects all the observation and
+will receive the actions. In production it will also contain a neuronal network. The `Agent` is the API for the _game code_.
+For this project we will use agents, that will ask the neuronal network for a decision, when won't have continuous 
+decision making.
+
+== Communication Requirements
+
+This section will set the requirements for the communication between the three essentials parts _game code_, 
+_unity ml-agents_ und `BrainAcademy`.
+
+=== Game Code to Unity ML Agents
+
+This is probably the most import part as will have the most influence on if the AI is going to work. The _game code_ 
+not make any direct request or information sharing other than calling for a decision. This will kept the agents much more
+flexible, so it will be possible to changes agents quickly and develop different agents for different solutions.
+
+So therefor the _game code_ should provide information sharing methods. The `CollectObservation()` should be able to ask
+for a lot of different information. E.g "Give me all enemies." The cropping of the data should happen in the `BrainAcademy`.
+It should be possible to ask for everything. This way training speed would increase drastically as when won't need new 
+Unity builds, but could change it easily in the `BrainAcademy`. This means for an enemy it should provide a public accessor
+which wraps the enemy into a data object.
+
+=== Unity ML Agents to BrainAcademy
+
+One of the crucial parts of developing an neuronal network with reinforcement learning is the input it gets. In order
+to keep the size small and enable fast decision making we decided not use any visual observation. In addition to that
+the *input for ONE network must always have the same dimension* meaning the neuronal network needs have same choices every
+time no matter if this choice is valid or possible. So the _game code_ needs to validate the decision. So the neuronal
+network can learn during training, that this decision has no impact and will learn not use it.
+
+There is another approach which catches invalid decisions before hand, but this would only increase the data send from
+and forth between all the parts, so this might not be an option, but could be reconsider during training.
```

Comment:
Which other approach?

Reply:
I'll could remember, but you were talking something about table before the neuronal network.